Source/Particles/Collision/MCCScattering.H

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2021 Roelof Groenewald
+ *
+ * This file is part of WarpX.
+ *
+ * License: ????
+ */
+#ifndef WARPX_PARTICLES_COLLISION_MCC_SCATTERING_H_
+#define WARPX_PARTICLES_COLLISION_MCC_SCATTERING_H_
+
+#include "Utils/WarpXConst.H"
+
+#include <AMReX_Random.H>
+
+/** \brief Function to perform elastic scattering of a particle in the lab
+ * frame. The particle velocities transformed to the COM frame where a hard
+ * sphere collision occurs. The resulting particle velocities are transformed
+ * back to the lab frame and the input particle's velocity is updated.
+ * @param[in/out] ux, uy, uz colliding particle's velocity
+ * @param[in] uCOM_x, uCOM_y, uCOM_z velocity of the center of momentum frame.
+ * @param[in] engine random number generator.
+ */
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void ElasticScattering ( amrex::ParticleReal& ux, amrex::ParticleReal& uy,
```

Comment:
Does the WarpX coding style prefer a file per class, such that there would be separate `ElasticScattering.H` and `BackScattering.H`?

Reply:
Possibly, but it seems to be that the functionality in individual scattering classes is too simple to justify breaking into a new file, for example the whole backscattering class only contains 3 lines. I'd say let's keep it as is until someone complains when we try to push it upstream.